sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/config/ThroughputControlGroupInternal.java

```diff
@@ -0,0 +1,116 @@
+package com.azure.cosmos.implementation.throughputControl.config;
+
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public abstract class ThroughputControlGroupInternal {
+    private final String groupName;
+    private final String id;
+    private final boolean isDefault;
+    private final CosmosAsyncContainer targetContainer;
+    private final Integer targetThroughput;
+    private final Double targetThroughputThreshold;
+
+    public ThroughputControlGroupInternal(
+        String groupName,
+        CosmosAsyncContainer targetContainer,
+        Integer targetThroughput,
+        Double targetThroughputThreshold,
+        boolean isDefault) {
+
+        checkArgument(StringUtils.isNotEmpty(groupName), "Group name can not be null or empty");
+        checkNotNull(targetContainer, "Target container can not be null");
+        checkArgument(targetThroughput == null || targetThroughput > 0, "Target throughput should be greater than 0");
+        checkArgument(
+            targetThroughputThreshold == null || (targetThroughputThreshold > 0 && targetThroughputThreshold <= 1),
+            "Target throughput threshold should between (0, 1]");
+
+        this.groupName = groupName;
+        this.targetContainer = targetContainer;
+        this.targetThroughput = targetThroughput;
+        this.targetThroughputThreshold = targetThroughputThreshold;
+        this.isDefault = isDefault;
+
+        this.id = this.targetContainer.getDatabase().getId() + "." + this.targetContainer.getId() + "." + this.groupName;
```

Comment:
'.' is a character allowed in resource names (Database / Collection) - I would consider using a separator that is not allowed in resource names - like '/' instead.

Reply:
GroupId will be part of the Id of the config items and control items, so '/' will not work here, what about () or [], some rare ones customer will use?

Reply:
I think you need to encode the Id you calculate anyway the restrictions on ID are too constrained for the group name I guess (id basically is optimized to allow only characters allowed in Uris). so {datebaseid}/{collectionId}/{encpdedGroupName} where the group name is encoded in Url-safe-Base64 (Base64.getUrlEncoder.encodeToString) or Url encoding

Base64 is probably the safest - but I woudl stick with Url encoding because that allows easier identification of which documents belong to which group when looking at the container with persisted documents. (this could very well also be seen as a benefit that base64 makes it harder because customers will be less encouraged to mock around with the documents :-))

Reply:
Thanks Fabian, this is a great idea, will use Url encoding for now. Will add in the next iteration.

Reply:
Actually UrlEncoding does not work for gateway mode, and my suspicious is gateway might decode it before sending request. 
So I changed to use Base64Encoding. The good thing currently, there is /group property in the containerItem, which will give us a pretty good hint about which container, which group this belongs to.